genet/modify/change_log.py

```diff
@@ -4,7 +4,7 @@
 from typing import Union, List
 
 
-class ChangeLog:
+class ChangeLog(pd.DataFrame):
```

Comment:
so when I started genet I knew I wanted to log changes but I wasn't sure how to do it. I settled on a dataframe because it's simple and most people know how to use pandas. At this point I don't think the changelog will ever not be a dataframe, it works so there is no point changing it. The only thing that might change is a couple of columns might get deleted because of redundancy one day. The `ChangeLog` had literally just one attribute `log` and that was to access the dataframe. I think it makes sense for it to be a dataframe, but I do get your point that it makes it difficult to change it. Users don't really interact with this object tbh, they can view it but it's mostly in the background for insurance